src/datalayer/store/promiseMiddleware.js

```diff
@@ -24,23 +26,27 @@ export default ({ dispatch, getState }) => (next) => (action) => {
 
   return p.then((result) => {
     // Dispatch successAction
-    next({
-      type: successAction,
-      payload: result,
-      extraPayload: rest.payload,
-    });
+    // If we are dispatching Login
+    if (successAction === 'ON_LOGIN_SUCCESS') {
+      Auth.setAuth(result.data.token, result.data.userId)
+        .then(() => {
+          next({
+            type: successAction,
+            payload: result,
+            extraPayload: rest.payload,
+          });
+        });
+    } else {
+      next({
+        type: successAction,
+        payload: result,
+        extraPayload: rest.payload,
+      });
+    }
```

Comment:
Use another middleware to handle `successAction === 'ON_LOGIN_SUCCESS'`

Reply:
You have helped me fix this